### PR TITLE
Ported Lunos' Gracidea.

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Title.

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

**Comment by ExpoSeed** [comment]
does this mean we are adding support for forme changes rather than leaving it to ultima's branch? i'm not sure which forme stuff is in our scope and which isn't

**Comment by AsparagusEduardo** [comment]
> does this mean we are adding support for forme changes rather than leaving it to ultima's branch? i'm not sure which forme stuff is in our scope and which isn't

Ultima's branch has Form IDs which is a whole other can of worms to handle. I don't see the harm on implementing form changes that don't require scripting (like Rotom's box of appliances), specially in this case as I feel Item Expansion should be the one implementing the feature as it's based on using the item itself. Shaymin Sky reverting back to Land should be handled by Pokémon Expansion in my opinion.

**Comment by ultima-soul** [comment]
So were sorta deciding a while ago, the following distinctions:
Item-based form changes in Item Expansion
Battle-based form changes in Battle Engine Upgrade
Everything else possibly Pokémon Expansion (I guess this would mainly be script-based ones)

**Review by ultima-soul — CHANGES_REQUESTED** [review]
Functions well.

File: src/party_menu.c [hunk 1]
```diff
@@ -5269,6 +5274,53 @@ void ItemUseCB_EvolutionStone(u8 taskId, TaskFunc task)
     }
 }
 
+void ItemUseCB_Gracidea(u8 taskId, TaskFunc task)
+{
+    #if defined SPECIES_SHAYMIN && defined SPECIES_SHAYMIN_SKY
+        struct Pokemon *mon = &gPlayerParty[gPartyMenu.slotId];
+        u16 item = gSpecialVar_ItemId;
+        u16 species = GetMonData(mon, MON_DATA_SPECIES);
+        u16 newSpecies;
```
[the remark this answers is not in the document]
Comment (LOuroboros) [reply src/party_menu.c:5283]: How about "forme" instead? :eyes:

**Comment by AsparagusEduardo** [comment]
Ready for re-review.

File: src/party_menu.c [hunk 2]
```diff
@@ -409,6 +412,9 @@ static bool8 SetUpFieldMove_Dive(void);
 #include "data/pokemon/tutor_learnsets.h"
 #include "data/party_menu.h"
 
+// Text string printed when changing the form of certain species like Shaymin and Giratina
+const u8 ChangedForm[] = _("{STR_VAR_1} changed Forme!{PAUSE_UNTIL_PRESS}");
```
Comment (LOuroboros) [inline src/party_menu.c:416]: This text string was updated somewhere inbetween Gen. 4 and Gen. 7. In [the latter](https://youtu.be/jQBBXX6c6VA?t=173), the string when changing either Shaymin's or Giratina's form actually says "Pokémon transformed!".

File: src/party_menu.c [hunk 3]
```diff
@@ -5269,6 +5275,63 @@ void ItemUseCB_EvolutionStone(u8 taskId, TaskFunc task)
     }
 }
 
+void ItemUseCB_FormChangeItem(u8 taskId, TaskFunc task)
+{
+    u16 item = gSpecialVar_ItemId;
+    switch (item)
+    {
+        case ITEM_GRACIDEA:
+            TryChangeShayminForm(taskId,task);
```
Comment (LOuroboros) [inline src/party_menu.c:5284]: Shouldn't there be an empty space inbetween `taskId` and `task`? :eyes:

**Comment by AsparagusEduardo** [comment]
Re-re-ready for Re-re-review

**Review by ExpoSeed — APPROVED** [review]
this is fine as is

**Review by ghoulslash — APPROVED** [review]

**Comment by ExpoSeed** [comment]
On hold until we decide form table shenanigans

**Comment by AsparagusEduardo** [comment]
Superseded by #1375

**Pull request closed without merging**